infra/charts/feast-feature-server/templates/deployment.yaml

```diff
@@ -33,19 +33,40 @@ spec:
           env:
             - name: FEATURE_STORE_YAML_BASE64
               value: {{ .Values.feature_store_yaml_base64 }}
-          command: ["feast", "serve", "-h", "0.0.0.0"]
+          command:
+            {{- if eq .Values.feast_mode "offline" }}
+            - "feast"
+            - "serve_offline"
+            - "-h"
+            - "0.0.0.0"
+            {{- else if eq .Values.feast_mode "ui" }}
+            - "feast"
+            - "ui"
+            - "-h"
+            - "0.0.0.0"
+            {{- else if eq .Values.feast_mode "registry" }}
+            - "feast"
+            - "serve_registry"
+            - "-h"
+            - "0.0.0.0"
+            {{- else }}
+            - "feast"
+            - "serve"
+            - "-h"
+            - "0.0.0.0"
+            {{- end }}
           ports:
-            - name: http
-              containerPort: 6566
+            - name: {{ .Values.feast_mode | default "http" }}
+              containerPort: {{ .Values.service.port }}
```

Comment:
i worry this won't work as port 80 generally requires root access and these containers won't have that at runtime. this is also not the intention of this helm value. this helm value is used to set the incoming service port... NOT the incoming container port.

instead, each "mode" should probably have its own port declaration/condition that's not configurable, but is set for the user depending on the chosen mode. for example -
```
online = 6566
ui = 8888
offline = nil?
registry = ?
```

Reply:
where we can add these ports within service section 
for example 
```
service:
  type: ClusterIP
  port: 80
  online : 6566
  ui : 8888
  offline : 8815
  registry : 6570
```

Reply:
by using the same type of conditionals in the `command` section -
e.g. 
```yaml
            {{- if eq .Values.feast_mode "offline" }}
```

Reply:
in service, these ports would be used for `targetPort`